packages/network-controller/src/NetworkController.ts

```diff
@@ -182,6 +182,8 @@ export class NetworkController extends BaseControllerV2<
 
   private trackMetaMetricsEvent: (event: MetaMetricsEventPayload) => void;
 
+  private previousProvider: { type: NetworkType; id?: string };
```

Comment:
1. Although we have properties in this controller that use the `private` keyword, we're trying to migrate off of it — what do you think about using `#` instead of `private`?

2. It appears that based on the way that this is used, either `setActiveNetwork` (which takes a network configuration ID) or `setProviderType` (which takes a provider "type") will get called. Also, the provider config `id` property will only get set when `setActiveNetwork` is set. So, what do you think about having this be either a NetworkType or a string? You could even type alias `string` to `NetworkConfigurationId` for clarity.

3. Since this property doesn't hold a provider object, and it doesn't hold a provider config object either, should we come up with another name for this property? Maybe something like a "network specifier":

``` suggestion
  #previousNetworkSpecifier: NetworkType | NetworkConfigurationId | null;
```

Reply:
Yeah I think this could make sense!

Reply:
done here https://github.com/MetaMask/core/pull/1132/commits/5fb32845402506720c09adebda05989ffa86e791